From: Ariel Elior <ariele@broadcom.com>
Date: Thu, 20 Sep 2012 05:26:41 +0000
Subject: [PATCH 31/77] bnx2x: remove false warning regarding interrupt number
Patch-mainline: v3.6-rc7
Git-commit: 185d4c8bf579322e1c2835d70729bc30f6f80f55
References: bnc#790597 FATE#313915

Since version 7.4 the FW configures in the pci config space the max
number of interrupts available to the physical function, instead of
the exact number to use.
This causes a false warning in driver when comparing the number of
configured interrupts to the number about to be used.

Signed-off-by: Ariel Elior <ariele@broadcom.com>
Signed-off-by: Eilon Greenstein <eilong@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Yuval Mintz <yuvalmin@broadcom.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
@@ -9830,6 +9830,16 @@ static void __devinit bnx2x_get_igu_cam_
 		}
 	}
 
+#ifdef CONFIG_PCI_MSI
+	/* Due to new PF resource allocation by MFW T7.4 and above, it's
+	 * optional that number of CAM entries will not be equal to the value
+	 * advertised in PCI.
+	 * Driver should use the minimal value of both as the actual status
+	 * block count
+	 */
+	bp->igu_sb_cnt = min_t(int, bp->igu_sb_cnt, igu_sb_cnt);
+#endif
+
 	if (igu_sb_cnt == 0)
 		BNX2X_ERR("CAM configuration error\n");
 }
